src/runtime/virtcontainers/network.go

```diff
@@ -233,18 +233,16 @@ type Network interface {
 	GetEndpointsNum() (int, error)
 }
 
-func generateVCNetworkStructures(ctx context.Context, network Network) ([]*pbTypes.Interface, []*pbTypes.Route, []*pbTypes.ARPNeighbor, error) {
-	if network.NetworkID() == "" {
-		return nil, nil, nil, nil
-	}
```

Comment:
I think you cannot just simply remove it, because this changes the original logic. I’d like to suggest adding a new argument, `enable_netns bool`. If VFIO is enabled, or in some scenarios where you don't want to use netns, set the value to `false`.

```
if enable_netns && network.NetworkID() == "" {
	return nil, nil, nil, nil
}
```

Reply:
good point. Because `generateVCNetworkStructures()` is wrapped by `setupNetworks()` function now, I added the old check back to `setupNetworks()`.